### DISCUSSION TITLE
introducing-the-microsoft-featuremanagement-library-adding-feature-flags-to-an-asp-net-core-app-part-1/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#127] [Announcements]
# introducing-the-microsoft-featuremanagement-library-adding-feature-flags-to-an-asp-net-core-app-part-1/

In this post I give a brief introduction to the preview Microsoft.FeatureManagement library and how to use it in an ASP.NET Core app

https://andrewlock.net/introducing-the-microsoft-featuremanagement-library-adding-feature-flags-to-an-asp-net-core-app-part-1/

**Comment by milyas-salik** [comment]
Hi Andrew,

Can I use this library (Microsoft.FeatureManagement) for other than .Net Core Applications such as ASP.NET MVC , ASP.NET Web APPLICATION, Console Application, Windows Application built on .NET.

If Yes can you provide some examples/steps to carry on or link to refer.

I have Solution built on .NET which as web applications, web apis and windows application built on .NET.

Thanks.

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at June 12 2019, 10:02, in reply to [Aida Crone](https://disqus.com/by/ovanpsiba/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/127#discussioncomment-6056619)</em>

---

<p>Thanks! As <a href="https://disqus.com/by/francesccastells/">Francesc Castells</a> points out, the main reason I don't use this is to avoid issues with refactoring implicitly changing the constant values. That said, in some cases it doesn't matter, in which case I'm with you :)</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at June 12 2019, 10:04, in reply to [Stilgar](https://disqus.com/by/NaibStilgar/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/127#discussioncomment-6056622)</em>

---

<p>No, the configuration system is not immutable. As shown in this post, you can change files on the fly, and the IConfiguration is updated. You can create a database config provider (example in the docs here <a href="https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-2.2#custom-configuration-provider)" rel="nofollow noopener" title="https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-2.2#custom-configuration-provider)">https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-2.2#custom-configuration-provider)</a>.</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at June 12 2019, 12:35, in reply to [Alex Dresko](https://disqus.com/by/alexdresko/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/127#discussioncomment-6056628)</em>

---

<p>The library has some more interesting features beyond the basics shown in this post, such as integration points in the MVC filter pipeline, and IFeatureFilters. I'll be covering those in later posts :)</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at June 17 2019, 11:25, in reply to [Mathieu Viales](https://disqus.com/by/mathieuviales/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/127#discussioncomment-6056635)</em>

---

<p>Fair enough, I prefer the latter (for the reasons you stated), so I'm not going to argue 😂</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at May 20 2020, 08:14, in reply to [Win Pooh](https://disqus.com/by/winpooh/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/127#discussioncomment-6056646)</em>

---

<p>You can inject the feature manager into Configure() but not into ConfigureServices()</p>

[a comment by an automated account is not shown]
